### DISCUSSION TITLE
Cannot decrypt file stream - readMessage(): Error: Misformed armored text

### DISCUSSION THREAD

**Discussion by jeff-pal** [openpgpjs/openpgpjs#1764] [Q&A]
openpgp: ^5.11.1
@types/openpgp: ^4.4.22
@openpgp/web-stream-tools: 0.0.11
Node.js: v16.20.2

I'm trying to decrypt a file but the `readMessage()` fails reading a encrypted file stream `Error: Misformed armored text` . 
It works to encrypt a stream and immediately decrypt it, as shown at [this example](https://github.com/openpgpjs/openpgpjs?tab=readme-ov-file#encrypt-and-decrypt-string-data-with-pgp-keys), but if it has to be stored and then decrypted it doesn't work.

```
Error: Misformed armored text
    at /Users/username/my-project/node_modules/openpgp/src/encoding/armor.js:253:25
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

How can we make it works to decrypt from an encrypted file stream?

I created a code sample which you can run to test this issue: https://github.com/jeff-pal/openpgp-debug/blob/main/src/index.ts

**Comment by larabr** [comment]
See https://github.com/openpgpjs/openpgpjs/discussions/1305#discussioncomment-741674 .

**Comment by jeff-pal** [comment]
Okay, so just to clarify the root cause of this issue in order to help other understanding this let me share my conclusion. According to the message format expected in readMessage it must be `Stream<string>` format if it's a stream.

```
export function readMessage<T extends MaybeStream<string>>(options: { armoredMessage: T, config?: PartialConfig }): Promise<Message<T>>;
export type MaybeStream<T extends Data> = T | Stream<T>;
```
make sense?

thanks @larabr !

**Reply by twiss** [reply]
If you're passing an `armoredMessage` it must be a `string` or `Stream<string>`; if you're passing a `binaryMessage` it must be a `Uint8Array` or `Stream<Uint8Array>` :)
